backend/protocol_rpc/endpoints.py

```diff
@@ -346,7 +347,10 @@ async def get_contract_schema_for_code(
         msg_handler=msg_handler.with_client_session(get_client_session_id()),
         contract_snapshot_factory=None,
     )
-    return json.loads(await node.get_contract_schema(contract_code))
+    schema = await node.get_contract_schema(contract_code)
+    if isinstance(schema, dict):
```

Comment:
@kp2pml30 shouldn't this validation be:
```suggestion
    if not isinstance(schema, dict):
```